File: 6_myLS_ver2/myLS.py
import os
import sys

def printStarsFunction():
    for i in range(80):
        print("*", end ='')

    print();
def openDirectoryFunction(path):
    return os.listdir(path)

printStarsFunction();
print("STARTING ls SIMULATOR. (print files inside given folder)")
printStarsFunction();

# Getting value from args or using dot

directoryFilesToBePrintedOut = None
if len(sys.argv) >= 2:
    if sys.argv[1] != None:
        directoryFilesToBePrintedOut = sys.argv[1]
    else:
        directoryFilesToBePrintedOut = '.'

#Opening directory
filesListInsideDir = openDirectoryFunction(directoryFilesToBePrintedOut);
for item in filesListInsideDir:
        print("Item: {0:30s}       ".format(item))

File: 1_myLS/myLS.pl
#!/usr/bin/perl
use Getopt::Std;

sub printStarsFunction {
    my @b = (1..40);
    for(@b){
        print "*";
    }

            print "\n";
  }

sub openDirectoryFunction{
    my $path = shift;
    opendir(my $handler, $path) || die "Can't open directory $directoryFilesToBePrintedOut";
    return $handler;
}


printStarsFunction();
print "STARTING ls SIMULATOR. (print files inside given folder)\n";
printStarsFunction();

# Getting value from args or using dot

$directoryFilesToBePrintedOut;
if(@ARGV)
{
    $directoryFilesToBePrintedOut = @ARGV[0];
}
else
{
    $directoryFilesToBePrintedOut = '.';
}

#Opening directory
my $handler = openDirectoryFunction($directoryFilesToBePrintedOut);
@filesList = sort(readdir($handler));
closedir $handler;

foreach $item ( @filesList ) 
{
    # The stat() function gets status information about a specified file 
    my $size = (stat("$directoryFilesToBePrintedOut/$item"))[7];
    my $mode = (stat("$directoryFilesToBePrintedOut/$item"))[2];
    $result = sprintf("Item: %30s   ||  size [ %08d ] bytes || permissions [ %04o ]     ", $item,  $size, $mode & 07777);
    print("$result \n");
}

